tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,340 @@
+              output of "docker exec <container_name> supervisorctl status". Program name
+              at here represents a program which is part of group <group_name>
+    @return: A dictionary where keys are the program names and values are their running
+             status and pids
+    """
+    group_program_info = defaultdict(list)
+    program_name = None
+    program_status = None
+    program_pid = -1
+
+    program_list = duthost.shell("docker exec {} supervisorctl status".format(container_name))
+    for program_info in program_list["stdout_lines"]:
+        if program_info.find(group_name) != -1:
+            program_name = program_info.split()[0].split(':')[1].strip()
+            program_status = program_info.split()[1].strip()
+            if program_status in ["EXITED", "STOPPED", "STARTING"]:
+                program_pid = -1
+            else:
+                program_pid = int(program_info.split()[3].strip(','))
+
+            group_program_info[program_name].append(program_status)
+            group_program_info[program_name].append(program_pid)
+
+    return group_program_info
+
+
+def get_program_info(duthost, container_name, program_name):
+    """
+    @summary: Get program running status and its pid by analyzing the command
+              output of "docker exec <container_name> supervisorctl status"
+    @return:  Program running status and its pid
+    """
+    program_status = None
+    program_pid = -1
+
+    program_list = duthost.shell("docker exec {} supervisorctl status".format(container_name))
+    for program_info in program_list["stdout_lines"]:
+        if program_info.find(program_name) != -1:
+            program_status = program_info.split()[1].strip()
+            if program_status == "RUNNING":
+                program_pid = int(program_info.split()[3].strip(','))
+            break
+
+    if program_pid != -1:
+        logger.info("Found program '{}' in the '{}' state with pid {}"
+                    .format(program_name, program_status, program_pid))
+
+    return program_status, program_pid
+
+
+def get_container_autorestart_states(duthost):
+    """
+    @summary: Get container names and their autorestart states by analyzing
+              the command output of "show container feature autorestart"
+    @return:  A dictionary where keys are the names of containers which have the
+              autorestart feature implemented and values are the autorestart feature
+              state for that container
+    """
+    container_autorestart_states = {}
+
+    show_cmd_output = duthost.shell(CMD_CONTAINER_FEATURE_AUTORESTART)
+    for line in show_cmd_output["stdout_lines"]:
+        container_name = line.split()[0].strip()
+        container_state = line.split()[1].strip()
+        if container_state in ["enabled", "disabled"]:
+            container_autorestart_states[container_name] = container_state
+
+    return container_autorestart_states
```

Comment:
This function can be moved to common/devices.py as a shared function if there is no such member function yet.

Reply:
Will moved this function to common/devices.py by creating a separate PR.